docs/tutorials/aws/RAG_with_DeepSeek_R1_model_on_Bedrock.md

```diff
@@ -0,0 +1,412 @@
+# Topic
+
+
+This tutorial introduces how to build RAG in Amazon managed OpenSearch with [DeepSeek R1 model](https://huggingface.co/deepseek-ai/DeepSeek-R1).
+If you are not using Amazon OpenSearch, you can refer to [deepseek_connector_chat_blueprint](https://github.com/opensearch-project/ml-commons/blob/main/docs/remote_inference_blueprints/deepseek_connector_chat_blueprint.md).
+
+Note: You should replace the placeholders with prefix `your_` with your own value
+
+
+# Steps
+
+## 0. Preparation
+
+### Deploy DeepSeek R1 model to Bedrock
+
+Follow this [notebook](https://github.com/DennisTraub/deepseekr1-on-bedrock/blob/main/deepseek-bedrock.ipynb) to deploy DeepSeek R1 model to Bedrock.
+
+Note the Bedrock DeepSeek R1 model ARN. This will be used in following steps.
+
+### Create OpenSearch Cluster
+
+Go to AWS OpenSearch console UI and create OpenSearch domain.
+
+Note the domain ARN and URL. They will be used in following steps.
+
+## 1. Create IAM role to invoke Sagemaker model
+To invoke Bedrock model, we need to create an IAM role with proper permission.
+This IAM role will be configured in connector. Connector will use this role to invoke DeepSeek model deployed on Bedrock.
+
+Go to IAM console, create IAM role `my_invoke_bedrock_deepseek_model_role` with:
+
+- Custom trust policy:
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "es.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+```
+- Permission
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "bedrock:InvokeModel"
+            ],
+            "Effect": "Allow",
+            "Resource": "your_DeepSeek_R1_model_ARN_"
+        }
+    ]
+}
+```
+
+Note the role ARN. It will be used in following steps.
+
+## 2. Configure IAM role in OpenSearch
+
+### 2.1 Create IAM role for Signing create connector request
+
+Generate a new IAM role specifically for signing your create connector request.
+
+
+Create IAM role `my_create_bedrock_deepseek_connector_role` with 
+- Custom trust policy. Note: `your_iam_user_arn` is the IAM user which will run `aws sts assume-role` in step 3.1
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "your_iam_user_arn"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+```
```

Comment:
````suggestion
```
Note: `your_iam_user_arn` is the IAM user which will run `aws sts assume-role` in step 3.1.
````